src/core/unit.c

```diff
@@ -3465,6 +3457,21 @@ int unit_patch_contexts(Unit *u) {
                         if (ec->protect_home == PROTECT_HOME_NO)
                                 ec->protect_home = PROTECT_HOME_READ_ONLY;
                 }
+
+                /* Finally patch no_new_privileges_implied if necessary */
+                if (!ec->no_new_privileges_implied &&
+                    (ec->syscall_whitelist ||
+                     !set_isempty(ec->syscall_filter) ||
+                     !set_isempty(ec->syscall_archs) ||
+                     ec->address_families_whitelist ||
+                     !set_isempty(ec->address_families) ||
+                     ec->memory_deny_write_execute ||
+                     ec->restrict_realtime ||
+                     exec_context_restrict_namespaces_set(ec) ||
+                     ec->protect_kernel_tunables ||
+                     ec->protect_kernel_modules ||
+                     ec->private_devices))
+                        ec->no_new_privileges_implied = true;
```

Comment:
what's the benefit of having an explicit variable for this? i mean, it should be fine to always calculate this on-demand. Caching this in a variable just creates the risk that it might get out-of-date.

(And no, performance is not an issue here, this is not going to be slow anyway...)